From f1c57b23dc1fa12bad08b0f3451894d5b92c99b8 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@gmail.com>
Date: Tue, 20 Nov 2018 14:06:55 +0800
Subject: [PATCH 04/25] drm: introduce helper for accessing EDID blob in
 drm_connector structure

commit 4a531879a607d9867038a6b0544a154f866c8b53 from
  https://github.com/analogdevicesinc/linux.git

There is a bunch of drivers that directly access edid_blob_ptr in
drm_connector_structure. This requires a cast and possibly a NULL ptr
check.

Wrap this in a common helper.

Signed-off-by: Andrea Merello <andrea.merello@gmail.com>
Signed-off-by: Lars-Peter Clausen <lars@metafoo.de>
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 include/drm/drm_edid.h | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/include/drm/drm_edid.h b/include/drm/drm_edid.h
index e3c404833115..52141c2edbbd 100644
--- a/include/drm/drm_edid.h
+++ b/include/drm/drm_edid.h
@@ -25,6 +25,7 @@
 
 #include <linux/types.h>
 #include <linux/hdmi.h>
+#include <drm/drm_crtc.h>
 
 struct drm_device;
 struct i2c_adapter;
@@ -443,6 +444,21 @@ static inline u8 drm_eld_get_spk_alloc(const uint8_t *eld)
 	return eld[DRM_ELD_SPEAKER] & DRM_ELD_SPEAKER_MASK;
 }
 
+/**
+ * drm_connector_get_edid - Get current EDID from the given connector
+ * @connector: pointer to the connector stucture
+ *
+ * This is a helper for accessing the drm blob buffered in the connector
+ * struct (if any)
+ */
+static inline struct edid *drm_connector_get_edid(struct drm_connector *connector)
+{
+	if (!connector->edid_blob_ptr)
+		return NULL;
+
+	return (struct edid *)connector->edid_blob_ptr->data;
+}
+
 /**
  * drm_eld_get_conn_type - Get device type hdmi/dp connected
  * @eld: pointer to an ELD memory structure
-- 
2.17.0

